### PR TITLE
implement thumbs_up and thumbs_down reaction

### PR DESCRIPTION
**Pull request by Badiru27**
Thumbs up and Thumbs down reaction to LLM Message

### CODE REVIEW

File: engine/client/client.go [hunk 1]
```diff
@@ -183,6 +195,69 @@ func (c *Client) handleMessage(message []byte) error {
...
+
+	selectedReaction := selectedHistory.Reaction
+	userReaction := msg.Payload.UserReaction
+
+	if selectedReaction != nil && *selectedReaction == userReaction {
+		selectedHistory.Reaction = nil
+	} else {
+		selectedHistory.Reaction = &userReaction
+	}
+
```
Comment (flygerian) [inline engine/client/client.go:251]: Too much whitespace

File: ui/src/pages/chats/ChatHistoryItem.svelte [hunk 2]
```diff
@@ -1,29 +1,64 @@
...
+  import Icon from '@iconify/svelte';
+  import { ChatHistoryDto, Reaction } from '@/dto/chat';
+  import { formatDistanceToNow, fromUnixTime } from 'date-fns';
+    import { reactToLLMMessage } from '@/api/ws'
+  export let historyItem: ChatHistoryDto;
+  export let align: 'left' | 'right' = 'left';
+  let showReactions = false;
+  let thumbsUp = historyItem.reaction === 'thumbs_up';
+  let thumbsDown = historyItem.reaction === 'thumbs_down';
+  const handleReaction = (reaction:Reaction) => {
```
Comment (flygerian) [inline ui/src/pages/chats/ChatHistoryItem.svelte:11]: Add some space here. Section is a bit unreadable

File: engine/client/client.go [hunk 3]
```diff
@@ -183,6 +195,69 @@ func (c *Client) handleMessage(message []byte) error {
...
+		razzle.Logger.Errorf("chat not found")
+		return fmt.Errorf("chat not found")
+	}
+
+	history := chat.History()
+	historyLength := len(history)
+	historyId := msg.Payload.HistoryId
+	var selectedHistory domain.ChatHistory
+	
+	for i := 0; i < historyLength; i++ {
```
Comment (flygerian) [inline engine/client/client.go:225]: I think this approach might fail in the coming months, but for now it's okay. We would definitely need to refactor it tho.

**Review by flygerian — CHANGES_REQUESTED** [review]

File: engine/client/client.go (hunk 3 above)
Comment (Badiru27) [reply engine/client/client.go:225]: Yeah, really weird having to loop through since I already have the historyId

**Review by flygerian — APPROVED** [review]

**Pull request merged**